### DISCUSSION TITLE
Allow customisation of `floating-ui` options

### DISCUSSION THREAD

**Discussion by alburkerk** [ReactTooltip/react-tooltip#909] [Ideas]
Hi, thank you for the great work en react-tooltip V5.

After reading some of the code, I found that the positioning in v5 is done with the amazing floating-ui library, great choice ! 

However, middleware array passed to floating-ui computation is hardcoded in react-tooltip package, allowing no configuration. Is there any plan to open up this configuration to the external API of react-tooltip. 

I understand that it would need to expose the internal API of floating-ui, demanding more maintenance work, but there are some concrete use cases which cannot be done with react-tooltip such as using the size optimizer or a most-space strategy, and those can be enabled very easily.

I could create a PR for that feature if the idea suits you

**Comment by gabrieljablonski** [comment]
We had this exact issue before, which led to the creation of the `positionStrategy` prop on the tooltip, which gets drilled down to `floating-ui`.

I'm not sure which would be best: creating a single tooltip prop with the object containing all of the relevant `floating-ui` props; or doing the same as with `positiongStrategy`. Both have pros and cons, but I lean towards listing the props individually, for the sake of simplicity when using the tooltip.

It would be great if you could do a draft PR, and we'll discuss further ideas there.

**Reply by alburkerk** [reply]
Thank you for you fast response, I created a [PR](https://github.com/ReactTooltip/react-tooltip/pull/912)  demonstrating the use case. 

I'm open for discussions on how to continue the implementation
